src/apps/lwaftr/lwaftr.lua

```diff
@@ -43,6 +44,43 @@ local function compute_binding_table_by_ipv4(binding_table)
    return ret
 end
 
+local function port_to_psid(port, reserved_ports_bit_count, psid_len)
+   local shift = 16 - (reserved_ports_bit_count + psid_len)
+   return band(rshift(port, shift), (2^psid_len)-1)
```

Comment:
"Be careful". :D
More pragmatically, I'd be tempted to do something like define INVALID_PSID=0xFFFFFFFF or something, and use that.